sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java

```diff
@@ -219,29 +219,50 @@ private void setSessionToken(ResourceId resourceId, String collectionName, Strin
     }
 
     private void addSessionToken(ResourceId resourceId, String partitionKeyRangeId, ISessionToken parsedSessionToken) {
-        this.collectionResourceIdToSessionTokens.compute(
-                resourceId.getUniqueDocumentCollectionId(), (k, existingTokens) -> {
-                    if (existingTokens == null) {
-                        ConcurrentHashMap<String, ISessionToken> tokens = new ConcurrentHashMap<>();
-                        tokens.put(partitionKeyRangeId, parsedSessionToken);
-                        return tokens;
+        ConcurrentHashMap<String, ISessionToken> existingTokensIfAny = this.collectionResourceIdToSessionTokens.get(resourceId.getUniqueDocumentCollectionId());
+
+        if (existingTokensIfAny != null) {
+            // if an entry for this collection exists, no need to lock the outer ConcurrentHashMap.
+            ConcurrentHashMap<String, ISessionToken>  existingTokens = existingTokensIfAny;
+            existingTokens.merge(partitionKeyRangeId, parsedSessionToken, (existingSessionTokens, newSessionToken) -> {
```

Comment:
Line 227 - 237 code is duplicated below in line 252 - 262.
To make the code better in readability, can we extract this piece of code in its own helper method ?

Reply:
good suggestion. Thanks. addressed in 048bec85ce063b82dc76c07ac0b26a18e6b5e0dd